// @ng-dynamic-forms/ui-material/esm5/src/radio-group/dynamic-material-radio-group.component.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    }
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output, ViewChild } from "@angular/core";
import { FormGroup } from "@angular/forms";
import { MatRadioGroup } from "@angular/material";
import { DynamicFormControlComponent, DynamicFormLayoutService, DynamicFormValidationService, DynamicRadioGroupModel } from "@ng-dynamic-forms/core";
var DynamicMaterialRadioGroupComponent = /** @class */ (function (_super) {
    __extends(DynamicMaterialRadioGroupComponent, _super);
    function DynamicMaterialRadioGroupComponent(layoutService, validationService) {
        var _this = _super.call(this, layoutService, validationService) || this;
        _this.layoutService = layoutService;
        _this.validationService = validationService;
        _this.blur = new EventEmitter();
        _this.change = new EventEmitter();
        _this.customEvent = new EventEmitter();
        _this.focus = new EventEmitter();
        return _this;
    }
    DynamicMaterialRadioGroupComponent.decorators = [
        { type: Component, args: [{
                    selector: "dynamic-material-radio-group",
                    template: "<ng-container [formGroup]=\"group\">\n\n    <mat-radio-group #matRadioGroup\n                     [formControlName]=\"model.id\"\n                     [id]=\"elementId\"\n                     [labelPosition]=\"model.getAdditional('labelPosition', 'after')\"\n                     [name]=\"model.name\"\n                     [ngClass]=\"[getClass('element', 'control'), getClass('grid', 'control')]\"\n                     (blur)=\"onBlur($event)\"\n                     (change)=\"onChange($event)\"\n                     (focus)=\"onFocus($event)\">\n\n        <mat-radio-button *ngFor=\"let option of model.options$ | async\"\n                          [name]=\"model.name\"\n                          [ngClass]=\"getClass('element', 'option')\"\n                          [value]=\"option.value\"><span [innerHTML]=\"option.label\"></span></mat-radio-button>\n\n    </mat-radio-group>\n\n</ng-container>\n"
                }] }
    ];
    /** @nocollapse */
    DynamicMaterialRadioGroupComponent.ctorParameters = function () { return [
        { type: DynamicFormLayoutService },
        { type: DynamicFormValidationService }
    ]; };
    DynamicMaterialRadioGroupComponent.propDecorators = {
        group: [{ type: Input }],
        layout: [{ type: Input }],
        model: [{ type: Input }],
        blur: [{ type: Output }],
        change: [{ type: Output }],
        customEvent: [{ type: Output }],
        focus: [{ type: Output }],
        matRadioGroup: [{ type: ViewChild, args: ["matRadioGroup",] }]
    };
    return DynamicMaterialRadioGroupComponent;
}(DynamicFormControlComponent));
export { DynamicMaterialRadioGroupComponent };
if (false) {
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.group;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.layout;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.model;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.blur;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.change;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.customEvent;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.focus;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.matRadioGroup;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.layoutService;
    /** @type {?} */
    DynamicMaterialRadioGroupComponent.prototype.validationService;
}
