// @ng-dynamic-forms/core/esm5/src/model/misc/dynamic-form-control-relation.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
/** @type {?} */
export var DYNAMIC_FORM_CONTROL_ACTION_DISABLE = "DISABLE";
/** @type {?} */
export var DYNAMIC_FORM_CONTROL_ACTION_ENABLE = "ENABLE";
/** @type {?} */
export var DYNAMIC_FORM_CONTROL_ACTION_VISIBLE = "VISIBLE";
/** @type {?} */
export var DYNAMIC_FORM_CONTROL_ACTION_HIDDEN = "HIDDEN";
/** @type {?} */
export var DYNAMIC_FORM_CONTROL_CONNECTIVE_AND = "AND";
/** @type {?} */
export var DYNAMIC_FORM_CONTROL_CONNECTIVE_OR = "OR";
/**
 * @record
 */
export function DynamicFormControlRelation() { }
if (false) {
    /** @type {?} */
    DynamicFormControlRelation.prototype.id;
    /** @type {?|undefined} */
    DynamicFormControlRelation.prototype.status;
    /** @type {?|undefined} */
    DynamicFormControlRelation.prototype.value;
}
/**
 * @record
 */
export function DynamicFormControlRelationGroup() { }
if (false) {
    /** @type {?} */
    DynamicFormControlRelationGroup.prototype.action;
    /** @type {?|undefined} */
    DynamicFormControlRelationGroup.prototype.connective;
    /** @type {?} */
    DynamicFormControlRelationGroup.prototype.when;
}
